rasa/shared/nlu/training_data/training_data.py

```diff
@@ -650,6 +650,73 @@ def can_train_nlu_model(self) -> bool:
         ]
         return not any([len(lst) > 0 for lst in lists_to_check])
 
+    def divide_into_chunks(self, max_size: int = 2) -> List["TrainingDataChunk"]:
+        """Divides the training data into smaller chunks.
+
+        Each chunk should be a good representation of the complete dataset. E.g. it
+        should not just contain examples of one intent, but instead match the
+        distribution of the complete dataset.
+
+        Args:
+            max_size: The maximum size (in MB) of one chunk.
+
+        Returns:
+            A list of all training data chunks.
+        """
+        pass
+
+
+class TrainingDataChunk:
+    """Holds a portion of the complete TrainingData."""
+
+    def __init__(
+        self,
+        training_examples: List[Message],
+        responses: Optional[Dict[Text, List[Dict[Text, Any]]]] = None,
+    ) -> None:
+        """Initialize a training data chunk.
+
+        Args:
+            training_examples: List of messages representing the training data.
+            responses: List of responses.
+        """
+        self.training_examples = training_examples
+        self.responses = responses
```

Comment:
`TrainingDataChunk` is used during featurization. Before that, we call the method `prepare_for_partial` training, which gets the complete dataset. If a featurizer needs regexes, etc. it can get it from there.

Reply:
Hmm, okay so we will still use the original `TrainingData` object for data loading. `TrainingDataChunk` is just for passing chunks through the pipeline. 👍

Reply:
Yes.